features/telemetry/stream-listener/src/main/java/org/opennms/netmgt/telemetry/listeners/StreamParser.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2020 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2020 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.listeners;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.opennms.netmgt.telemetry.api.receiver.Parser;
+
+import io.netty.buffer.ByteBuf;
+
+public interface StreamParser extends Parser {
+
+    /**
+     * Invoked when Interface is added/deleted.
+     */
+    Config getConfig(Integer nodeId, String ipAddress);
+
+    /**
+     * Invoked initially when stream listener is starting
+     */
+    List<Config> getMatchingConfig();
+
+
+    CompletableFuture<?> parse(final ByteBuf buffer,
+                               final InetSocketAddress remoteAddress) throws Exception;
+}
```

Comment:
This is still to much bound to OpenConfig. Can we push the connection part even further to the parser? I imagine the following: The Listener is responsible for identifyng which connections should exit whereas the parser is responsible for opening and closing the connections as dictated by the listener. A proposal for a according interface:

```suggestion
public interface StreamParser extends Parser {
    public interface Connection extends Closeable {}

    public Connection connect(final int nodeId, final String ipAddress) throws Exception;
}
```

The listener then has to keep track of connection instances.